### ISSUE TITLE
Connection configuration allows multiple definitions

### ISSUE THREAD

**Issue by mikary** [aspnet/EntityFramework#1391]
When a connection string is defined with the DbContext "ConnectionString" and "ConnectionStringKey" keys, the last defined connection is used. This definition is ambiguous and should probably throw.

```cs
    public class MyContext : DbContext
    {
        public MyContext(IServiceProvider serviceProvider)
            : base(serviceProvider)
        {
        }
    }
```

```cs
        public void Main(string[] args)
        {
            var configuration = new Configuration();
            configuration.AddJsonFile("config.json");

            var serviceProvider = new ServiceCollection()
                .AddEntityFramework(configuration)
                .AddSqlServer()
                .AddDbContext<MyContext>()
                .ServiceCollection
                .BuildServiceProvider();

            using (var context = new MyContext(serviceProvider))
            {
                context.Database.EnsureCreated();
                Console.WriteLine(context.Database.AsRelational().Connection.ConnectionString);
                Console.ReadLine();
            }
        }
```

```json
{
    "Data": {
        "DefaultConnection": {
            "ConnectionString": "Data Source=(localdb)\\v11.0;Initial Catalog=MyDBSecond;Integrated Security=True;MultipleActiveResultSets=True;Connect Timeout=30"
        }
    },

    "EntityFramework": {
        "MyContext": {
            "ConnectionString": "Data Source=(localdb)\\v11.0;Initial Catalog=MyDBFirst;Integrated Security=True;MultipleActiveResultSets=True;Connect Timeout=30",
            "ConnectionStringKey": "Data:DefaultConnection:ConnectionString"
        }
    }
}
```

**Comment by smitpatel** [comment]
The Key, value data is stored in a dictionary inside DbContextOptions.  We are not checking if the key already exist in the dictionary so values are overwritten and the last one inserted is used. Adding a check if the options contains key or not would fix this.

**Comment by rowanmiller** [comment]
We should ditch the two key names and just support the 'name=xyz' syntax in the connection string value

**Comment by smitpatel** [comment]
fixed via 4d6c82826674edfee69ee98c6ee1970ee5e89ee5
